components/event-sources/apis/sources/v1alpha1/http_types.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2019 The Kyma Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	duckv1beta1 "knative.dev/pkg/apis/duck/v1beta1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// HTTPSource is a HTTP event source.
+type HTTPSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   HTTPSourceSpec   `json:"spec,omitempty"`
+	Status HTTPSourceStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// HTTPSourceList is a list of HTTPSource.
+type HTTPSourceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []HTTPSource `json:"items"`
+}
+
+// HTTPSourceSpec defines the desired state of HTTPSource.
+type HTTPSourceSpec struct {
+}
```

Comment:
On the ticket there was a proposal for a spec. Will there be a second PR to fill in the missing pieces?

Reply:
We agreed to add them as part of the adapter PR, because they are irrelevant without the adapter.

Reply:
i see, then we should move the discussion about the spec to the other ticket, so that it does not get resolved without implementation.

Reply:
It won't, otherwise the adapter won't work :)

Reply:
i meant moving the discussion to the adapter implementation ticket #6159. This way all changes related to the adapter implementation would be in one place

Reply:
so no code changes required, just some housekeeping

Reply:
Done